src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -42,7 +42,7 @@ impl fmt::Display for ConstraintCategory {
         // Must end with a space. Allows for empty names to be provided.
         match self {
             ConstraintCategory::Assignment => write!(f, "assignment "),
-            ConstraintCategory::Return => write!(f, "return "),
+            ConstraintCategory::Return => write!(f, "returning this value "),
```

Comment:
I like how this reads, though it's less "concise", which I think some people value. Still, seems ok to me.